src/com/codeenginestudio/bookManagement/servlets/EditBookServlet.java

```diff
@@ -0,0 +1,51 @@
+package com.codeenginestudio.bookManagement.servlets;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.codeenginestudio.bookManagement.manage.ManageBook;
+import com.codeenginestudio.bookManagement.models.Book;
+import com.codeenginestudio.bookManagement.util.BookUtil;
+import com.codeenginestudio.bookManagement.validator.BookValidator;
+
+@WebServlet("/EditBookServlet")
+public class EditBookServlet extends HttpServlet {
+
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+
+		int bookId = Integer.parseInt(request.getParameter("bookId"));
+		request.setAttribute("book", ManageBook.getOneBook(bookId));
+		BookUtil._displayView(request, response, "/view/AddOrEditBook.jsp");
+	}
+
+	protected void doPost(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+
+		int bookId = Integer.parseInt(request.getParameter("bookId"));
+		String bookName = request.getParameter("bookName");
+		String bookAuthor = request.getParameter("bookAuthor");
+		List<Map<String, String>> bookInputErrors = BookValidator._validate(bookName);
+
+		if (!bookInputErrors.get(0).isEmpty()) {
+
+			request.setAttribute("book", new Book(bookId, bookName, bookAuthor));
+			request.setAttribute("bookErr", bookInputErrors);
+			BookUtil._displayView(request, response, "/view/AddOrEditBook.jsp");
+		} else {
+
+			ManageBook.editBook(bookId, bookName, bookAuthor);
+			request.setAttribute("listBooks", ManageBook.getListBooks());
+			BookUtil._displayView(request, response, "/view/Home.jsp");
```

Comment:
After editing a book, f5 on the browser and see what happens! Redirect can helps!